game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/aa/FiringGroupSplitterAa.java

```diff
@@ -0,0 +1,80 @@
+package games.strategy.triplea.delegate.battle.steps.fire.aa;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ACTIVE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ALIVE;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.TechAbilityAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.fire.FiringGroup;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Creates AA and Targeted Hit firing groups
```

Comment:
For the splitter class, it would be nice to see some more detail in the javadoc to explain what characteristics would be true of the output `List<FiringGroup>`. I'm not really sure what "Targeted Hit firing groups" are.

Perhaps we can/should add 'targeted hit' to the glossary?

Reply:
The terms are broadly not well defined. TripleA was created before the concept of Domain-Driven-Design was more mainstream, as such instead of defining and using domain terminology, brand new terminology was invented (EG: "Node").